core/trino-main/src/main/java/io/trino/sql/ExpressionUtils.java

```diff
@@ -265,6 +276,40 @@ public static Expression filterConjuncts(Metadata metadata, Expression expressio
         };
     }
 
+    /**
+     * Returns whether expression is effectively literal. An effectitvely literal expression is a constant value, or null,
+     * in either {@link Literal} form, or other form returned by {@link LiteralEncoder}.
```

Comment:
Maybe add a note to underline that this does not apply to all constants, but only to constants of certain shapes?

Reply:
added `other constant expressions like a deterministic function call with constant arguments are not considered effectitvely literal` and replaced `constant` with `simple constant` in the second sentence.